test/sqllogictest/transform/join_index.slt

```diff
@@ -198,8 +194,7 @@ where foo.a = bar.a
 %3 =
 | Join %0 %1 %2 (= #0 #2) (= #3 #4)
 | | implementation = Differential %0.(#0) %1.(#0) %2.(#0)
-| | demand = (#0, #3, #5)
-| Filter !(isnull(#0)), !(isnull(#3))
+| Filter !(isnull(#0)), !(isnull(#2)), !(isnull(#3)), !(isnull(#4))
```

Comment:
Is this a bad change? It feels like it is an interplay with the `Demand` analysis which would have put a `Project` in place to reduce the needed columns down.